src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -79,16 +79,28 @@ public AWSLoggerCore(AWSLoggerConfig config, string logType)
             _config = config;
             _logType = logType;
 
-            var credentials = DetermineCredentials(config);
-
-            if (_config.Region != null)
-            {
-                _client = new AmazonCloudWatchLogsClient(credentials, Amazon.RegionEndpoint.GetBySystemName(_config.Region));
-            }
-            else
-            {
-                _client = new AmazonCloudWatchLogsClient(credentials);
-            }
+            if (!string.IsNullOrEmpty(_config.ServiceUrl))
+            {
+                var awsConfig = new AmazonCloudWatchLogsConfig() { ServiceURL = _config.ServiceUrl };
+                if (_config.ServiceUrl.StartsWith("http://", StringComparison.OrdinalIgnoreCase))
+                    awsConfig.UseHttp = true;
+                if (_config.Credentials != null)
+                    _client = new AmazonCloudWatchLogsClient(_config.Credentials, awsConfig);
+                else
+                    _client = new AmazonCloudWatchLogsClient("aws", "aws", awsConfig);

```

Comment:
What is the point of passing in "aws" for access key and secret key?

Reply:
Recall that someone complained that `AmazonCloudWatchLogsClient` would not work if provided empty-strings or null. Even if authentication is not enabled.

Reply:
I'm uncomfortable putting that in, it seems very random based on some tools requirement. Seems like the user could setup a profile with "aws" if they really need it.

Reply:
Lost my google-skills, so cannot find the original user that complained that credentials fails when not providing random non-empty values. Maybe because of this:

https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Core/Amazon.Runtime/Credentials/BasicAWSCredentials.cs#L45